src/Database/LSMTree/Model/Normal.hs

```diff
@@ -51,10 +53,28 @@ class SomeSerialisationConstraint a where
     -- Note: cannot fail.
     deserialise :: BS.ByteString -> a
 
+-- Note: we might add extra requirements,
+-- as this instance may produce arbitrary large serialised values.
+-- Do we need to limit maximum serialised size?
+-- Or even require it to be constant (and known in advance?)
 instance SomeSerialisationConstraint BS.ByteString where
     serialise = id
     deserialise = id
 
+-- | MSB, so order is preserved.
+instance SomeSerialisationConstraint Word64 where
+    serialise w = BS.pack [b1,b2,b3,b4,b5,b6,b7,b8] where
+        b8 = fromIntegral $        w    .&. 0xff
+        b7 = fromIntegral $ shiftR w  8 .&. 0xff
+        b6 = fromIntegral $ shiftR w 16 .&. 0xff
+        b5 = fromIntegral $ shiftR w 24 .&. 0xff
+        b4 = fromIntegral $ shiftR w 32 .&. 0xff
+        b3 = fromIntegral $ shiftR w 40 .&. 0xff
+        b2 = fromIntegral $ shiftR w 48 .&. 0xff
+        b1 = fromIntegral $ shiftR w 56 .&. 0xff
+
+    deserialise = BS.foldl' (\acc d -> acc * 0x100 + fromIntegral d) 0
```

Comment:
We should add an optimisation TODO here. We should be able to use some primops to load and endian-swapping primops.

One thing we'll decide later when we do the file format details is if we guarantee keys to have aligned addresses or not.

Reply:
Note: this is *model* `SomeSerializationConstraint`.

First you (and @jorisdral) have to agree/decide that we want or don't want to have only one serialization constraint.